### ISSUE TITLE
Training continue_data.py gives Runtimerror

### ISSUE THREAD

**Issue by janlycka** [brownvc/deep-synth#8]
Greetings to all the developers of deep-synth,

I am experiencing some trouble with commit cc0e991, possibly due to me using newer libraries that it's designed for. I would like to know which

12.01.2021

OS: Ubuntu 18.04
CUDA: 10.1
Python: 3.8.5
pytorch: 1.7.1 (with 0.4 I cannot even run `python create_data.py`)
scipy: 1.2.0

Having successfully generated the data with create_data.py, we now have several folders in the data directory: bedroom, living, office

Running `location_train.py --data-dir bedroom --save-dir res_2_bedroom --train-size 500 --use-count` works

Running `continue_train.py --data-dir bedroom --save-dir res_1_bedroom --train-size 500 --use-count` doesn't

I get the following Error:
`RuntimeError: Expected object of scalar type Long but got scalar type Bool for argument #2 'target' in call to _thnn_nll_loss_forward`

![image](https://user-images.githubusercontent.com/18504309/104327900-8f1ca900-54eb-11eb-98b8-c2b909d49bd6.png)

looking forward to hearing from you,
KR

**Comment by kwang-ether** [comment]
probably a version issue - just update the binary target to Long

**Comment by janlycka** [comment]
I have very limited knowledge of pytorch. Could you extrapolate?
Thanks & KR

**Comment by kwang-ether** [comment]
https://github.com/brownvc/deep-synth/blob/b800e11290b763b58e7d3b30329769a7b77cd12a/deep-synth/continue_dataset.py#L67 change this to long

**Comment by janlycka** [comment]
Alright, I've solved it. (Edit: Also thanks for your reply, I was typing up the response at the time at which you answered)

**TLDR**:
in continue_train.py, change lines 137 and 179 from
 `data, target = data.cuda(), target.cuda()`
to
`data, target = data.cuda(), target.cuda().long()`

continue_train.py now runs properly
`python continue_train.py --data-dir bedroom --save-dir res_1_bedroom --train-size 500 --use-count`

I hadn't understood your answer linguistically, as I had been unaware of the existence of tensors, which may have **Boolean** type, **Double** type or **Long** type. This is something different to the standard c-family datatypes bool, double, int, long etc. If I hear update bool to long, I understand casting

This Project uses a very old version of the [pytorch library](https://pytorch.org/get-started/locally/) 0.4.1, which was [impossible ](https://pytorch.org/get-started/previous-versions/#via-pip)to use together with [CUDA](https://blogs.nvidia.com/blog/2012/09/10/what-is-cuda-2/) 10.1 which we have installed on our system. Since 2018, the time of release of deep-synth, pytorch developers have changed things about with pytorch and at present, pytorch 1.7.1 expects to receive tensors with the type Long as opposed to Bool, as it's previously been, which we now must cast like so:
`target = target.cuda().long()`

This helpful [link ](https://discuss.pytorch.org/t/how-to-cast-a-tensor-to-another-type/2713) and two pythonists whom I know saved the day

A similar issue for further reading can be found [here](https://stackoverflow.com/questions/56510189/how-to-convert-torch-int64-to-torch-longtensor?fbclid=IwAR1rgc770bVHJu0-lV_vVd7C-2d61_DAGdnaQ-SBmqTQsF6eqC-vZgA6-uQ#56510366)

whilst on the note of the use of outdated libraries in this commit, which is understandable considering it's from 2018, whoever else may be trying to get this thing afloat in the future, make sure that your [SciPy](https://github.com/scipy/scipy/releases) library is 1.2 or older, as there has as well been a [breaking change](https://stackoverflow.com/questions/57545125/attributeerror-module-scipy-misc-has-no-attribute-toimage) in version 1.3

### libraries links

[Matplotlib](https://github.com/matplotlib/matplotlib/releases)
[Numpy](https://github.com/numpy/numpy/releases)
[Numba](https://github.com/numba/numba/releases)
[SciKit-learn](https://github.com/scikit-learn/scikit-learn/releases)
[SciPy](https://github.com/scipy/scipy/releases)

### Versions of the libraries for commit cc0e991
Here are the versions of the libraries contained in requirements.txt, which I currently have installed on my machine and which most definitely work

replace [requirements.txt](https://github.com/brownvc/deep-synth/blob/cc0e9910a742bf78f8466682d788ec34419213dc/deep-synth/requirements.txt) with this (gets you the desired versions)
```
matplotlib==3.3.3
numpy==1.19.2
numba==0.52.0
pybullet==2.1.3
pyquaternion==0.9.9
scikit-learn==0.24.0
scipy==1.2.0
```

install by running
`pip install -r requirements.txt`

### check the library versions

And finally, this serves to save you the 20 minutes of googling

install pyTorch like so
`conda install pytorch torchvision torchaudio cudatoolkit=10.1 -c pytorch`

check pytorch version
`python -c "import torch; print(torch.__version__)"`
**1.7.1**

check matplotlib version
`python -c "import matplotlib; print(matplotlib.__version__)"`
**3.3.3**

check numpy version
`"import numpy; print(numpy.version.version)"`
**1.19.2**

check scipy version
`python -c "import scipy; print(scipy.version.version)"`
**1.2.0**
